### PR TITLE
pdb2pqr: init at 3.6.1

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes
PDB2PQR - determining titration states, adding missing atoms, and assigning charges/radii to biomolecules.

homepage: https://www.poissonboltzmann.org/
source: https://github.com/Electrostatics/pdb2pqr

and add some python-module dependencies. ([mmcif-pdbx](https://github.com/Electrostatics/mmcif_pdbx), [propka](https://github.com/jensengroup/propka))

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 231095` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>pdb2pqr (python310Packages.pdb2pqr)</li>
    <li>pdb2pqr.dist (python310Packages.pdb2pqr.dist)</li>
    <li>python310Packages.mmcif-pdbx</li>
    <li>python310Packages.mmcif-pdbx.dist</li>
    <li>python310Packages.propka</li>
    <li>python310Packages.propka.dist</li>
    <li>python311Packages.mmcif-pdbx</li>
    <li>python311Packages.mmcif-pdbx.dist</li>
    <li>python311Packages.pdb2pqr</li>
    <li>python311Packages.pdb2pqr.dist</li>
    <li>python311Packages.propka</li>
    <li>python311Packages.propka.dist</li>
  </ul>
</details>

**Comment by onny** [comment]
When trying to run the binary I get
```
./results/pdb2pqr/bin/pdb2pqr
Traceback (most recent call last):
  File "/nix/store/p5fq0dgsla9m3y938dyx0yw1nch44163-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/__init__.py", line 23, in <module>
    from . import multiarray
  File "/nix/store/p5fq0dgsla9m3y938dyx0yw1nch44163-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/multiarray.py", line 10, in <module>
    from . import overrides
  File "/nix/store/p5fq0dgsla9m3y938dyx0yw1nch44163-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/b8acc3hrq1f7mqmm9qbpljsb2vh6nbjb-python3.10-pdb2pqr-3.6.1/bin/.pdb2pqr-wrapped", line 6, in <module>
    from pdb2pqr.main import main
  File "/nix/store/m9p5ffig3p6m64zrkszlzykdgli4awsb-python3.11-pdb2pqr-3.6.1/lib/python3.11/site-packages/pdb2pqr/__init__.py", line 10, in <module>
    from .main import main
  File "/nix/store/m9p5ffig3p6m64zrkszlzykdgli4awsb-python3.11-pdb2pqr-3.6.1/lib/python3.11/site-packages/pdb2pqr/main.py", line 20, in <module>
    from . import aa
  File "/nix/store/m9p5ffig3p6m64zrkszlzykdgli4awsb-python3.11-pdb2pqr-3.6.1/lib/python3.11/site-packages/pdb2pqr/aa.py", line 9, in <module>
    from . import residue
  File "/nix/store/m9p5ffig3p6m64zrkszlzykdgli4awsb-python3.11-pdb2pqr-3.6.1/lib/python3.11/site-packages/pdb2pqr/residue.py", line 9, in <module>
    from . import utilities as util
  File "/nix/store/m9p5ffig3p6m64zrkszlzykdgli4awsb-python3.11-pdb2pqr-3.6.1/lib/python3.11/site-packages/pdb2pqr/utilities.py", line 17, in <module>
    import numpy as np
  File "/nix/store/p5fq0dgsla9m3y938dyx0yw1nch44163-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/__init__.py", line 141, in <module>
    from . import core
  File "/nix/store/p5fq0dgsla9m3y938dyx0yw1nch44163-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.10 from "/nix/store/ysd0w2ffcn21snv39770gna78wcf524j-python3-3.10.11/bin/python3.10"
  * The NumPy version is: "1.24.2"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'
```
Is this intended?

**Comment by natsukium** [comment]
Thanks for checking.
It works fine on my "x86_64-linux" and "aarch64-darwin", but could you please tell me the environment where the error occurs?

**Comment by natsukium** [comment]
In this log, it looks like a strange thing is happening, calling python 3.11 libraries from a python 3.10 wrapper.🤔
```
Traceback (most recent call last):
  File "/nix/store/b8acc3hrq1f7mqmm9qbpljsb2vh6nbjb-python3.10-pdb2pqr-3.6.1/bin/.pdb2pqr-wrapped", line 6, in <module>
    from pdb2pqr.main import main
  File "/nix/store/m9p5ffig3p6m64zrkszlzykdgli4awsb-python3.11-pdb2pqr-3.6.1/lib/python3.11/site-packages/pdb2pqr/__init__.py", line 10, in <module>
    from .main import main
```

**Comment by onny** [comment]
> Thanks for checking. It works fine on my "x86_64-linux" and "aarch64-darwin", but could you please tell me the environment where the error occurs?

I'm using nixpkgs-review so this could be the case, thanks for testing. I'll do a manual check

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -7242,6 +7244,8 @@ self: super: with self; {
 
   pcpp = callPackage ../development/python-modules/pcpp { };
 
+  pdb2pqr = callPackage ../development/python-modules/pdb2pqr { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:7247]: Has this a lib part? Otherwise this entry should be removed.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please add pythonImportsCheck

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/top-level/python-packages.nix:7247]: Yes, `pdb2pqr` also has a python interface that can be extended.

**Comment by natsukium** [comment]
Thanks for your review.
Added `pythonImportsCheck`.

I think `pythonImportsCheck` is redundant because it passes `pytest`. Should I still check it in this case?

**Comment by SuperSandro2000** [comment]
> I think `pythonImportsCheck` is redundant because it passes `pytest`. Should I still check it in this case?

Yes, it should always be added.

**Comment by natsukium** [comment]
I understand, I will remember to check it in the future.

**Comment by natsukium** [comment]
@SuperSandro2000 san, I forgot to push and the `pythonImportsCheck` was not added.
Would you please check again?

**Pull request merged**